packages/type-extrinsics/src/utils/createExtrinsic.ts

```diff
@@ -0,0 +1,51 @@
+// Copyright 2017-2018 @polkadot/extrinsics authors & contributors
+// This software may be modified and distributed under the terms
+// of the ISC license. See the LICENSE file for details.
+
+import { createType } from '@polkadot/api-codec/codec';
+import { FunctionMetadata } from '@polkadot/api-codec/Metadata';
+import { Extrinsic, Text } from '@polkadot/api-codec/index';
+import u8aConcat from '@polkadot/util/u8a/concat';
+
+import { ExtrinsicFunction } from '../types';
+
+/**
+ * From the metadata of a function in the module's storage, generate the function
+ * that will return the correctly-encoded Extrinsic.
+ *
+ * @param index - Index of the module section in the modules array.
+ */
+export default function createExtrinsic (
+  prefix: string | Text,
+  name: string | Text,
+  index: number,
+  meta: FunctionMetadata
+): ExtrinsicFunction {
+  let extrinsicFn: any;
+
+  // If the extrinsic function has an argument of type `Origin`, we ignore it
+  // FIXME should be `arg.type !== Origin`, but doesn't work...
+  const expectedArgs = meta.arguments.filter((arg) => arg.type.toString() !== 'Origin');
+
+  extrinsicFn = (...args: any[]): Extrinsic => {
+    if (expectedArgs.length.valueOf() !== args.length) {
+      throw new Error(`Extrinsic ${prefix}.${name} expects ${expectedArgs.length.valueOf()} arguments, got ${args.length}.`);
+    }
+
+    return new Extrinsic(
+      u8aConcat(
+        new Uint8Array([index, meta.id.toU8a()[1]]),
```

Comment:
Not sure if that's the cleanest way

Reply:
Well, we get to 3,0... so... Just not sure this is u16 (not sure why it is not a u8) and the toU8a() encodes to little-endian, so the value we are actually interested in is in position 0?

i.e.

`0x0004` -> `Uint8Array([4, 0])`

I would actually just `toNumber()` it there. It is supposed to fit in a byte, if it goes over 255 we have bigger issues.